pkg/hcp/telemetry/exporter_test.go

```diff
@@ -0,0 +1,155 @@
+package telemetry
+
+import (
+	"context"
+	"errors"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/go-uuid"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+)
+
+type fakeScraper struct {
+	scrapeCalled atomic.Bool
+	scrapeErr    error
+}
+
+func (s *fakeScraper) scrape(ctx context.Context, filters []string, labels pcommon.Map) (pmetric.Metrics, error) {
+	s.scrapeCalled.Store(true)
+
+	if s.scrapeErr != nil {
+		return pmetric.NewMetrics(), s.scrapeErr
+	}
+
+	return pmetric.NewMetrics(), nil
+}
+
+type fakeStateTracker struct {
+	runCalled      atomic.Bool
+	getStateCalled atomic.Bool
+	state          *state
+}
+
+func (s *fakeStateTracker) Run(ctx context.Context) {
+	s.runCalled.Store(true)
+}
+
+func (s *fakeStateTracker) GetState() (*state, bool) {
+	s.getStateCalled.Store(true)
+
+	if s.state != nil {
+		return s.state, true
+	}
+
+	return nil, false
+}
+
+type fakeClient struct {
+	exportCalled atomic.Bool
+	exportErr    error
+}
+
+func (e *fakeClient) ExportMetrics(ctx context.Context, metrics pmetric.Metrics) error {
+	e.exportCalled.Store(true)
+
+	if e.exportErr != nil {
+		return e.exportErr
+	}
+
+	return nil
+}
+
+func Test_Worker(t *testing.T) {
+	t.Parallel()
+
+	for name, tc := range map[string]struct {
+		modState  func(*state)
+		nilState  bool // nil state entirely
+		scrapeErr error
+		exportErr error
+
+		expectScrape bool // whether we expect to scrape metrics
+		expectExport bool // whether we expect to attempt to export metrics
+	}{
+		"success": {
+			expectScrape: true,
+			expectExport: true,
+		},
+		"disabled": {
+			modState: func(s *state) {
+				s.disabled = true
+			},
+		},
+		"disabled nil state": {
+			nilState: true,
```

Comment:
if you wanted to re-use modState for this case, you could make modState return a *state.